File: src/com/sjsu/cs146/homework/hw8/hw8.java
package com.sjsu.cs146.homework.hw8;
import java.util.*;

public class hw8 {
    public static void main(String[] args) {
        String s = "ab";
        System.out.println(longestPalindrome(s));
    }
    public static int longestPalindrome(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        int count = s.length();
        for(int i = 0; i < s.length(); i++){
            char j = s.charAt(i);
            if(map.containsKey(j)) map.put(j, map.get(j) + 1);
            else map.put(j, 1);
        }
        for (int i : map.values()) if(i % 2 == 1) count--;
        if(count < s.length()) count++;
        return count;
    }
}

File: src/com/sjsu/cs146/homework/hw8/hw8.py
class Solution:
    def longestPalindrome(self, s):
        letters = dict()
        count = len(s)
        for c in s:
            if c in letters.keys(): letters.update({c:letters.get(c) + 1})
            else: letters.update({c:1})
        for i in letters.values(): 
            if i % 2 == 1: count -= 1
        if count < len(s): count += 1
        return count

solution = Solution()
print(solution.longestPalindrome("speediskey"))